### PR TITLE
chore: update dependencies in submodules

### PR DESCRIPTION
**Pull request by justinsb**
Our submodules should also get auto-updated and auto-tidied by the
Makefile targets; this should then be done automatically by our github
action scheduled task.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -324,12 +324,16 @@ ko-kops-utils-cp-push:
...
-	cd tests/e2e; go mod tidy
 	cd hack; go mod tidy
+	cd tests/e2e; go mod tidy
+	cd tools/otel/traceserver; go mod tidy
 
 .PHONY: goget
 goget:
 	go get $(shell go list -f '{{if not (or .Main .Indirect)}}{{.Path}}{{end}}' -mod=mod -m all | grep -v spotinst-sdk-go)
+	cd hack; go get $(shell go list -f '{{if not (or .Main .Indirect)}}{{.Path}}{{end}}' -mod=mod -m all | grep -v spotinst-sdk-go)
+	cd tests/e2e; go get $(shell go list -f '{{if not (or .Main .Indirect)}}{{.Path}}{{end}}' -mod=mod -m all | grep -v spotinst-sdk-go)
```
Comment (hakman) [inline Makefile:335]: I don't think I want ginkgo and kubetest updated, at least not as part of this flow.
Also, we need to remove  `grep -v spotinst-sdk-go` everywhere.

**Review by hakman — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (justinsb) [reply Makefile]: Done; we don't import ginkgo directly anywhere, and we only import kubetest2 in one submodule; added the exclusion there.

**Review by hakman — APPROVED** [review]

**Pull request merged**